core/src/main/java/com/datastax/oss/driver/internal/core/session/RequestProcessor.java

```diff
@@ -46,4 +46,7 @@
       DefaultSession session,
       InternalDriverContext context,
       String sessionLogPrefix);
+
+  /** Builds a failed result to directly report the given error. */
+  ResultT newFailure(RuntimeException error);
```

Comment:
All existing processors need to implement this new method. I initially thought of adding ad-hoc code in every processor, but the advantage of this approach is that we can do things in an abstract manner in `DefaultSession`, so it's a better way to ensure that it will work for every processor.